src/net_processing.cpp

```diff
@@ -4704,7 +4704,7 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
 
                     bool fGotBlockFromCache = false;
                     {
-                        LOCK(cs_most_recent_block);
+                        LOCK(most_recent_block_mutex);
```

Comment:
> ... it is not possible that within one section another one is called...

It is not obvious where functions with a non-trivial body--like `PushMessage`--are called. Even everything is correct now, there is a possibility to introduce a recursion in he future by modifying the `PushMessage` function or other functions been called from within it.

Could we just call `PushMessage` without holding `most_recent_block_mutex`?

Reply:
Fair enough, will take a deeper look.